axum/src/extract/ws.rs

```diff
@@ -331,6 +340,11 @@ impl WebSocket {
     pub async fn close(mut self) -> Result<(), Error> {
         self.inner.close(None).await.map_err(Error::new)
     }
+
+    /// Return the selected WebSocket subprotocol, if one has been chosen
+    pub fn protocol(&self) -> Option<&str> {
```

Comment:
I think we should just return `&HeaderValue` and let the user deal with whether they care about non utf-8 or not.